src/insights_client/metrics.py

```diff
@@ -31,6 +37,27 @@ def _proxy_settings(rhsm_config):
     return {"https": proxy}
 
 
+def _is_offline(cfg):
+    '''
+    Determine whether offline mode was specified in
+    either the config or the CLI
+
+    :param cfg: RawConfigParser with the
+                insights-client config file read into it
+
+    Returns True if offline, False if not offline
+    '''
+    # look in config file first
+    try:
+        offline = cfg.getboolean("insights-client", "offline")
+    except configparser.NoOptionError:
+        offline = False
+    if "--offline" in sys.argv:
```

Comment:
Is it this simple? Shouldn’t we use argparser for consistency? I mean, is it possible that _--offline_ somehow appears in _argv_, but it is parsed differently? E.g. as a value to a different argument.

Reply:
```suggestion
    parser = ArgumentParser()
    parser.add_argument('--offline', default=False, action="store_true")
    args = parser.parse_args()
    if args.offline:
        offline = True
```